### PR TITLE
Create the resource Address groups

### PR DESCRIPTION
**Pull request by diogoEsteves42**
Adding network security `AddressGroups` resource support.

fixes: https://github.com/hashicorp/terraform-provider-google/issues/14299

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_security_address_group`
```

### CODE REVIEW

**Comment by SarahFrench** [comment]
Hi @diogoEsteves42 , sorry for not doing a proper review yet - I'm on bug hunting duty this week so I'm juggling a few things! I'll trigger the tests and give this PR a review once the tests complete 🚀

**Comment by diogoEsteves42** [comment]
Thanks for the message @SarahFrench, wish you can handle everything 💪🏼 

Ok about not merging the main.

**Comment by SarahFrench** [comment]
There's been an issue with the tests running on this PR - I'll run /gcbrun just in case it helps

**Comment by SarahFrench** [comment]
There's a problem with the tests at the moment - please don't push any more commit! Thanks

File: mmv1/products/networksecurity/AdressGroup.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits.
+      Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "labels"
+    description: |
+      Set of label tags associated with the AddressGroup resource.
+      An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+  - !ruby/object:Api::Type::Enum
+    name: "type"
```
Comment (SarahFrench) [inline mmv1/products/networksecurity/AdressGroup.yaml:105]: The API's docs say that this field is required - could you please add `required: true` in this section?

File: mmv1/products/networksecurity/AdressGroup.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    description: |
+      The type of the Address Group. Possible values are "IPV4" or "IPV6".
+    values:
+      - :IPV4
+      - :IPV6
+  - !ruby/object:Api::Type::Array
+    name: "items"
+    item_type: Api::Type::String
+    required: true
```
Comment (SarahFrench) [inline mmv1/products/networksecurity/AdressGroup.yaml:114]: From the API docs it seems this field is optional - could you please update this in the yaml?

https://cloud.google.com/traffic-director/docs/reference/network-security/rest/v1beta1/organizations.locations.addressGroups#AddressGroup.FIELDS.items

File: mmv1/third_party/terraform/tests/resource_network_security_address_groups_test.go.erb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+resource "google_network_security_address_group" "foobar" {
+    name        = "%s"
+    location    = "us-central1"
+    description = "my address groups. Update"
+    type        = "IPV4"
+    capacity    = "100"
+    labels      = {
+		foo = "foo"
+    }
+    items = ["208.80.154.224/32"]
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/tests/resource_network_security_address_groups_test.go.erb:69]: Sometimes we find that list ordering can be an issue with providers - could you please add another item to this array and place them 'out of order', e.g. ["208.80.15**5**.224/32", "208.80.15**4**.224/32"]. Thanks!

**Review by SarahFrench — COMMENTED** [review]
Thanks for bearing with me and the test issues - here's a review 😅 . I found a few small things that need attention but otherwise everything looks good!

File: mmv1/third_party/terraform/tests/resource_network_security_address_groups_test.go.erb (hunk 3 above)
Comment (diogoEsteves42) [reply mmv1/third_party/terraform/tests/resource_network_security_address_groups_test.go.erb]: Ohh, I'll pay attention to that, but I didn't see anything related. Tested locally and works, thanks.

File: mmv1/products/networksecurity/AdressGroup.yaml (hunk 1 above)
Comment (diogoEsteves42) [reply mmv1/products/networksecurity/AdressGroup.yaml:105]: Sure, thanks. Done

**Comment by diogoEsteves42** [comment]
Hi @SarahFrench, thank you for the review, about the field required and the other not, maybe I just switched them, appreciate the review.  About the test scenario I've added the item you just mentioned ran the test locally and it worked well.
Again, thanks for the review process.

**Comment by diogoEsteves42** [comment]
Hi @SarahFrench, everything ran with success, could we follow the review process, because I will need to update part of the example documentation of another resource with the usage of `addressGroup`.  Appreciate your attention

**Review by SarahFrench — APPROVED** [review]

**Pull request merged**